### PR TITLE
Add Google Keep Reader

### PR DESCRIPTION
**Pull request by pycui**
Reads notes from Google Keep. 
Note Google Keep's API only works if you have (1) Enterprise account (2) use Service Account to authenticate (3) a Domain-wide Delgation enabled for the service account with the Google Keep scopes - which requires a Google Workspace super admin to set up. Making it less useful for normal users.

### CODE REVIEW

**Comment by jerryjliu** [comment]
amazing! will take a look tomorrow @pycui

**Comment by pycui** [comment]
@jerryjliu maybe holding off a bit - after re-thinking about the limitations of current Google Keep API, I think it's not practical for most of users of Keep (the API frankly is not as productionized as Google Docs or Drives) . Would we be open to use "unofficial" API libraries like https://github.com/kiwiz/gkeepapi which actually works better for normal user?

File: llama_hub/google_keep/README.md
Comment (jerryjliu) [inline llama_hub/google_keep/README.md:28]: nit: *VectorStoreIndex (gpt prefix is deprecated)

**Review by jerryjliu — COMMENTED** [review]
> @jerryjliu maybe holding off a bit - after re-thinking about the limitations of current Google Keep API, I think it's not practical for most of users of Keep (the API frankly is not as productionized as Google Docs or Drives) . Would we be open to use "unofficial" API libraries (like https://github.com/kiwiz/gkeepapi which actually works better for normal user) in the hub?

In general yeah we'd be open to unofficial library support. The goal of llamahub is really to give users a diversity of choices in getting started (hence why we have 7+ different pdf loaders), so I'd just pick one to start with. honestly i'm down to review/land this if it looks ok (as long as it works)

**Comment by pycui** [comment]
> In general yeah we'd be open to unofficial library support. The goal of llamahub is really to give users a diversity of choices in getting started (hence why we have 7+ different pdf loaders), so I'd just pick one to start with. honestly i'm down to review/land this if it looks ok (as long as it works)

Sounds good I will switch to [gkeepapi](https://github.com/kiwiz/gkeepapi) API given it works better for most. Will rerequest review once ready.

**Comment by jerryjliu** [comment]
sounds good, thanks!

File: llama_hub/google_keep/README.md
Comment (pycui) [reply llama_hub/google_keep/README.md]: Done.

File: llama_hub/google_keep/README.md
Comment (jerryjliu) [inline llama_hub/google_keep/README.md:14]: let's do `from llama_hub.google_keep.base import GoogleKeepReader` (we try to do that by default now)

File: llama_hub/google_keep/base.py
Comment (jerryjliu) [inline llama_hub/google_keep/base.py:4]: currently we do lazy imports for tests to pass (not the best eng design decision, it might be worth thinking about a better way to do tests/linting even with optional imports

**Review by jerryjliu — COMMENTED** [review]
thanks! some comments. also not sure why poetry.lock is modified

**Review by jerryjliu — APPROVED** [review]

**Comment by pycui** [comment]
@jerryjliu can you or a maintainer help merge it? Thanks!

**Review by jerryjliu — APPROVED** [review]

**Pull request merged**
